saleor/product/filters.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import unicode_literals
+from collections import OrderedDict
+
+from django_filters import (FilterSet, MultipleChoiceFilter, RangeFilter,
+                            OrderingFilter)
+from django.forms import CheckboxSelectMultiple
+from django.utils.translation import pgettext_lazy
+
+from django_prices.models import PriceField
+
+from .models import Product, ProductAttribute
+
+
+DEFAULT_SORT = 'name'
+
+
+SORT_BY_FIELDS = [{'value': 'name',
+                   'label': pgettext_lazy('Sort by filter', 'name')},
+                  {'value': 'price',
+                   'label': pgettext_lazy('Sort by filter', 'price')}]
+
+
+class ProductFilter(FilterSet):
+    def __init__(self, *args, **kwargs):
+        self.category = kwargs.pop('category')
+        super(ProductFilter, self).__init__(*args, **kwargs)
+        self.product_attributes, self.variant_attributes = \
+            self._get_attributes(self.category)
+        self._add_product_attributes_filters()
+        self._add_product_variants_attributes_filters()
+        self.filters = OrderedDict(sorted(self.filters.items()))
```

Comment:
As a result of each of the three lines above the instance variable `self.filters` is modified. I think that instance variables should represent state of the instance and here we have to call these three particular lines to have it initialized properly. Instead, the first two functions should return some structures, then join them together, sort and and the end the result should be assigned to `self.filters`.

Reply:
functions now updates self.filters instead of modifing it